finality-aleph/src/network/gossip/service.rs

```diff
@@ -26,16 +34,16 @@ use crate::{
 /// 3. Outgoing messages, sending them out, using 1.2. to broadcast.
 pub struct Service<N: RawNetwork, D: Data> {
     network: N,
-    messages_from_user: mpsc::UnboundedReceiver<D>,
-    messages_for_user: mpsc::UnboundedSender<D>,
+    messages_from_user: mpsc::UnboundedReceiver<Command<D, N::PeerId>>,
+    messages_for_user: mpsc::UnboundedSender<(D, N::PeerId)>,
     authentication_connected_peers: HashSet<N::PeerId>,
     authentication_peer_senders: HashMap<N::PeerId, TracingUnboundedSender<D>>,
     spawn_handle: SpawnTaskHandle,
 }
 
-struct ServiceInterface<D: Data> {
-    messages_from_service: mpsc::UnboundedReceiver<D>,
-    messages_for_service: mpsc::UnboundedSender<D>,
+struct ServiceInterface<D: Data, P: Clone + Debug + Eq + Hash + Send + 'static> {
```

Comment:
Right now `ServiceInterface` just sends `Command` without any `Protocol`. And after that everything is just sent over `Authentication` protocol.

Maybe it would be nice to add a field to this struct that states which protocol to use. Then we can extend `Command` enum to also send `Protocol`. Something like:
```
Command::Send(Data, PeerId, Protocol)
```
This way we do not have to change `Network` trait to contain protocol in each sending call (broadcast, send, send_to_random). This way connection manager would not have to know about  `Protocol` enum. It would just have a network interface that sends through Authentication protocol. And from what I understand in the future we want to have several protocols (like Authentication and BlockSync).

Reply:
Yeah, good point, improved.